core/pkg/server/sender.go

```diff
@@ -554,6 +554,8 @@ func (s *Sender) sendRequestDefer(request *service.DeferRequest) {
 			if s.exitRecord != nil {
 				s.fileStream.FinishWithExit(s.exitRecord.GetExit().GetExitCode())
 			} else {
+				s.logger.CaptureError(
```

Comment:
should we also add a comment that in theory it shouldn't happen?

Reply:
IMO the CaptureError implies this